tests/test_kitsune.py

```diff
@@ -0,0 +1,337 @@
+                                    httpretty.Response(body=request_callback)
+                               ])
+
+
+class TestKitsuneBackend(unittest.TestCase):
+    """Kitsune backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        kitsune = Kitsune(KITSUNE_SERVER_URL, origin='test')
+
+        self.assertEqual(kitsune.url, KITSUNE_SERVER_URL)
+        self.assertEqual(kitsune.origin, 'test')
+        self.assertIsInstance(kitsune.client, KitsuneClient)
+
+        # When origin is empty or None it will be set to
+        # the value in url
+        kitsune = Kitsune(KITSUNE_SERVER_URL)
+        self.assertEqual(kitsune.url, KITSUNE_SERVER_URL)
+        self.assertEqual(kitsune.origin, KITSUNE_SERVER_URL)
+
+        kitsune = Kitsune(KITSUNE_SERVER_URL, origin='')
+        self.assertEqual(kitsune.url, KITSUNE_SERVER_URL)
+        self.assertEqual(kitsune.origin, KITSUNE_SERVER_URL)
+
+    def __check_questions_contents(self, questions):
+        self.assertEqual(questions[0]['data']['num_votes'], 2)
+        self.assertEqual(questions[0]['data']['num_answers'], 0)
+        self.assertEqual(questions[0]['data']['locale'], 'en-US')
+        self.assertEqual(questions[0]['origin'], KITSUNE_SERVER_URL)
+        self.assertEqual(questions[0]['uuid'], '8fa01e2aadf37c6f2fa300ce529fd1a23feac333')
+        self.assertEqual(questions[0]['updated_on'], 1467798846.0)
+        self.assertEqual(questions[0]['offset'], 0)
+
+        if len(questions) > 1:
+            self.assertEqual(questions[1]['data']['num_votes'], 1)
+            self.assertEqual(questions[1]['data']['num_answers'], 0)
+            self.assertEqual(questions[1]['data']['locale'], 'es')
+            self.assertEqual(questions[1]['origin'], KITSUNE_SERVER_URL)
+            self.assertEqual(questions[1]['uuid'], '7a3fbfb33cfaaa32b2f121994faf019a054a9a06')
+            self.assertEqual(questions[1]['updated_on'], 1467798439.0)
+            self.assertEqual(questions[1]['offset'], 1)
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether the questions are returned"""
+
+        HTTPServer.routes()
+
+        # Test fetch questions with their reviews
+        kitsune = Kitsune(KITSUNE_SERVER_URL)
+
+        questions = [question for question in kitsune.fetch()]
+
+        self.assertEqual(len(questions), 4)
+
+        self.__check_questions_contents(questions)
+
+    @httpretty.activate
+    def test_fetch_offset(self):
+        """Test whether the questions are returned offset"""
+
+        HTTPServer.routes()
+
+        # Test fetch questions with their reviews
+        kitsune = Kitsune(KITSUNE_SERVER_URL)
+
+        # Get all questions
+        offset = 0
+        questions = [question for question in kitsune.fetch(offset=offset)]
+        self.assertEqual(len(questions), 4)
+        self.__check_questions_contents(questions)
+
+        # 4 total questions minus 2, 2 questions returned
```

Comment:
I would like to check on these cases that the `offset` assigned to each item is right one.